move-vm-backend/tests/mock.rs

```diff
@@ -39,3 +40,28 @@ impl Storage for StorageMock {
         data.remove(key);
     }
 }
+
+pub struct SubstrateApiMock {
+    pub storage: StorageMock,
+}
+
+impl SubstrateAPI for SubstrateApiMock {
+    fn transfer(
+        &self,
+        _from: AccountAddress,
+        to: AccountAddress,
+        amount: u128,
+    ) -> Result<(), TransferError> {
+        Ok(self
+            .storage
+            .set(to.as_slice(), amount.to_be_bytes().as_slice()))
+    }
+
+    fn get_balance(&self, of: AccountAddress) -> u128 {
+        if let Some(b_data) = self.storage.get(of.as_slice()) {
+            u128::from_be_bytes(b_data.try_into().unwrap())
+        } else {
+            123
```

Comment:
nit: zero would be a _more default choice_ :sweat_smile:

Reply:
the idea is to have something non-default to make sure where it is coming from

Reply:
So returning that value is some kind of testing feature and will be changed in the future?

Reply:
it is a testing feature, and I dont see why it will be changed in the future.